data-pipeline-flink/assessment-aggregator/src/main/scala/org/sunbird/dp/assessment/functions/AssessmentAggregatorFunction.scala

```diff
@@ -83,56 +87,96 @@ class AssessmentAggregatorFunction(config: AssessmentAggregatorConfig,
+           * EX - For PDF content users are generating the ASSESS events
+           */
+          if (isNegligibleValue(totalQuestions)) {
+            context.output(config.failedEventsOutputTag, event)
+            metrics.incCounter(config.invalidContentSkippedCount)
+          } else {
+            /**
+             * If the totalQuestions from the content Meta and Events count are matching
+             * Then we are computing the score metrics and updating the table
+             */
+            if (totalQuestions == sortAndFilteredEvents.size) {
+              updateDB(scoreMetrics = computeScoreMetrics(sortAndFilteredEvents), event = event)(metrics, context)
+            } else {
+              /**
+               * if the totalQuestions Count value is not matching with the event size then job should skip those events
+               */
+              context.output(config.failedEventsOutputTag, event)
+              metrics.incCounter(config.ignoredEventsCount)
+            }
           }
+        } else {
+          /**
+           * If the config.skipMissingRecords = false then it will work as previous logic, no further validation
+           */
+          updateDB(scoreMetrics = computeScoreMetrics(sortAndFilteredEvents), event = event)(metrics, context)
         }
       } else {
         context.output(config.failedEventsOutputTag, event)
         metrics.incCounter(config.failedEventCount)
       }
     } catch {
       case ex: Exception =>
-        logger.info("Assessment Failed with exception :", ex)
+        ex.printStackTrace()
+        logger.info("Assessment Failed with exception :", ex.getMessage)
         event.markFailed(ex.getMessage)
         context.output(config.failedEventsOutputTag, event)
         metrics.incCounter(config.failedEventCount)
     }
   }
 
+  def updateDB(scoreMetrics: Aggregate, event: Event)(metrics: Metrics, context: ProcessFunction[Event, Event]#Context): Unit = {
+    val assessmentDBResult = getAssessment(event)
+    if (null == assessmentDBResult) {
+      saveAssessment(event, Aggregate(scoreMetrics.totalScore, scoreMetrics.totalMaxScore, scoreMetrics.grandTotal, scoreMetrics.questionsList), new DateTime().getMillis)
+      metrics.incCounter(config.dbUpdateCount)
+      metrics.incCounter(config.batchSuccessCount)
+      context.output(config.scoreAggregateTag, event)
+      createIssueCertEvent(event, context, metrics)
+    }
+    else {
+      metrics.incCounter(config.dbReadCount)
+      if (event.assessmentEts > assessmentDBResult.getTimestamp("last_attempted_on").getTime) {
+        saveAssessment(event, Aggregate(scoreMetrics.totalScore, scoreMetrics.totalMaxScore, scoreMetrics.grandTotal, scoreMetrics.questionsList),
+          assessmentDBResult.getTimestamp("created_on").getTime)
+        metrics.incCounter(config.dbUpdateCount)
+        metrics.incCounter(config.batchSuccessCount)
+        context.output(config.scoreAggregateTag, event)
```

Comment:
Merge attempt logic is missing here. In addition once the attempts are merged - the totalQuestionCount comparison should be redone again

Reply:
Adding this task as backlog, will pickup in the coming release.